src/sage/combinat/partition.py

```diff
@@ -5486,6 +5486,52 @@ def coloring(i):
                               immutable=True, multiedges=True)
         return self.dual_equivalence_graph(directed, coloring)
 
+    def specht_module(self, base_ring=None):
+        r"""
+        Return the Specht module corresponding to ``self``.
+
+        EXAMPLES::
+
+            sage: SM = Partition([2,2,1]).specht_module(QQ)
+            sage: SM
+            Specht module of [(0, 0), (0, 1), (1, 0), (1, 1), (2, 0)] over Rational Field
+            sage: s = SymmetricFunctions(QQ).s()
+            sage: s(SM.frobenius_image())
+            s[2, 2, 1]
+        """
+        from sage.combinat.specht_module import SpechtModule
```

Comment:
maybe factor out a function `specht_module_builder` that takes a ring, a size and cells ? Instead of doing these double imports everywhere ?

Reply:
I don't really see the benefit. To me, it adds more indirection and complexity to save on just doing one import at each point of the code instead of two (as the second import is still done). To me, it is equivalent to having
```python
class Bar:
    def __init__(self, x):
        self._x = x

def foo(x):
    from above import Bar
    return Bar(x)
```
which I would want to just call `Bar(x)` within my code.